src/test/java/com/google/sps/MutationTest.java

```diff
@@ -0,0 +1,539 @@
+    Assert.assertTrue(graphNodes.contains(gNodeA));
+    Assert.assertEquals(graphNodesMap.get("A"), gNodeA);
+    Assert.assertTrue(graph.hasEdgeConnecting(gNodeA, gNodeB));
+  }
+
+  /*
+   * Check that a deleting an edge between unconnected nodes has no impact
+   */
+  @Test
+  public void deleteAbsentEdgeSuccess() {
+    MutableGraph<GraphNode> graph = GraphBuilder.directed().build();
+    HashMap<String, GraphNode> graphNodesMap = new HashMap<>();
+    graph.addNode(gNodeA);
+    graph.addNode(gNodeB);
+    graph.addNode(gNodeC);
+    graph.putEdge(gNodeA, gNodeB);
+    graph.putEdge(gNodeB, gNodeC);
+
+    graphNodesMap.put("A", gNodeA);
+    graphNodesMap.put("B", gNodeB);
+    graphNodesMap.put("C", gNodeC);
+
+    Mutation removeAC =
+        Mutation.newBuilder()
+            .setType(Mutation.Type.DELETE_EDGE)
+            .setStartNode("A")
+            .setEndNode("C")
+            .build();
+
+    boolean success = servlet.mutateGraph(removeAC, graph, graphNodesMap);
+    Assert.assertTrue(success);
+
+    Set<GraphNode> graphNodes = graph.nodes();
+    Assert.assertEquals(graphNodes.size(), 3);
+    Assert.assertTrue(graphNodes.contains(gNodeA));
+    Assert.assertEquals(graphNodesMap.get("A"), gNodeA);
+    Assert.assertTrue(graphNodes.contains(gNodeB));
+    Assert.assertEquals(graphNodesMap.get("B"), gNodeB);
+    Assert.assertTrue(graphNodes.contains(gNodeC));
+    Assert.assertEquals(graphNodesMap.get("C"), gNodeC);
+
+    Assert.assertTrue(graph.hasEdgeConnecting(gNodeA, gNodeB));
+    Assert.assertTrue(graph.hasEdgeConnecting(gNodeB, gNodeC));
+  }
+
+  /*
+   * Check that adding tokens to a node has the right effect
+   */
+  @Test
+  public void addTokens() {
+    MutableGraph<GraphNode> graph = GraphBuilder.directed().build();
+    HashMap<String, GraphNode> graphNodesMap = new HashMap<>();
+    graph.addNode(gNodeA);
+    graph.addNode(gNodeB);
+    graph.addNode(gNodeC);
+    graph.putEdge(gNodeA, gNodeB);
+    graph.putEdge(gNodeB, gNodeC);
+    graphNodesMap.put("A", gNodeA);
+    graphNodesMap.put("B", gNodeB);
+    graphNodesMap.put("C", gNodeC);
+
+    List<String> newTokens = new ArrayList<>();
+    newTokens.add("1");
+    newTokens.add("2");
+    newTokens.add("3");
+
+    TokenMutation tokenMut =
+        TokenMutation.newBuilder()
+            .setType(TokenMutation.Type.ADD_TOKEN)
+            .addTokenName("1")
+            .addTokenName("2")
+            .addTokenName("3")
+            .build();
+
+    Mutation addTokenToA =
+        Mutation.newBuilder()
+            .setStartNode("A")
+            .setType(Mutation.Type.CHANGE_TOKEN)
+            .setTokenChange(tokenMut)
+            .build();
+
+    boolean success = servlet.mutateGraph(addTokenToA, graph, graphNodesMap);
+    Assert.assertTrue(success);
+
+    Set<GraphNode> graphNodes = graph.nodes();
+    Assert.assertTrue(graphNodesMap.containsKey("A"));
+    GraphNode newNodeA = graphNodesMap.get("A");
```

Comment:
Do you expect gnodeA to be reference equal to newNodeA? If so, also add an assert for this.